frontend/workflows/projectSelector/src/storage.tsx

```diff
@@ -0,0 +1,89 @@
+import _ from "lodash";
+
+import type { GroupState, ProjectsState, ProjectState, State } from "./types";
+import { Group } from "./types";
+
+const LOCAL_STORAGE_STATE_KEY = "dashState";
+
+/**
+ * Determines if an object is of type @type {ProjectState}.
+ */
+const isProjectState = (state: ProjectState | object): state is ProjectState => {
+  return (state as ProjectState).checked !== undefined;
```

Comment:
does `as` actually check at runtime?

i'm having trouble finding any docs about this

Reply:
`as` won't do any checking but it'll cast the object to `ProjectState` so we can further check the fields on the object without typescript complaining.

[This page](https://www.typescriptlang.org/docs/handbook/advanced-types.html) is out of date but this is the original documentation on type guards.

Reply:
but there are other fields, for example `custom` that are not checked here then, is that right?

Reply:
yes that's correct. Those fields are optional and therefore not expected to always be there.